core/iwasm/common/wasm_shared_memory.c

```diff
@@ -52,11 +53,17 @@ wasm_shared_memory_init()
 {
     if (os_mutex_init(&shared_memory_list_lock) != 0)
         return false;
-    /* wait map not exists, create new map */
+
+    if (os_mutex_init(&wait_map_lock) != 0) {
```

Comment:
The hash map already has its lock if it is created with `bh_hash_map_create(.., true, ..), and its related operations will be locked, could you check whether we should add the lock again?

Reply:
Thanks, I didn't know about that. I suppose I still need it since I'm also using it to make the removal from list and hashmap atomic.
https://github.com/eloparco/wasm-micro-runtime/blob/0c8da385a8dc696e46b4e93457da42da73cdc965/core/iwasm/common/wasm_shared_memory.c#L477

list removal:
https://github.com/eloparco/wasm-micro-runtime/blob/0c8da385a8dc696e46b4e93457da42da73cdc965/core/iwasm/common/wasm_shared_memory.c#L483
map removal:
https://github.com/eloparco/wasm-micro-runtime/blob/0c8da385a8dc696e46b4e93457da42da73cdc965/core/iwasm/common/wasm_shared_memory.c#L490
https://github.com/eloparco/wasm-micro-runtime/blob/0c8da385a8dc696e46b4e93457da42da73cdc965/core/iwasm/common/wasm_shared_memory.c#L382

So I think I can pass `false` to the constructor to avoid using the built-in lock and use the one I created instead.

Reply:
The wait_map lock was not necessary in the end, I pushed a change to get rid of it